src/tree_cmd.cpp

```diff
@@ -134,24 +140,47 @@ static void PlantTreesOnTile(TileIndex tile, TreeType treetype, uint count, uint
 static TreeType GetRandomTreeType(TileIndex tile, uint seed)
 {
 	switch (_settings_game.game_creation.landscape) {
-		case LT_TEMPERATE:
-			return (TreeType)(seed * TREE_COUNT_TEMPERATE / 256 + TREE_TEMPERATE);
-
-		case LT_ARCTIC:
-			return (TreeType)(seed * TREE_COUNT_SUB_ARCTIC / 256 + TREE_SUB_ARCTIC);
-
+		case LT_TEMPERATE: return TREE_RANDOM_TEMPERATE;
+		case LT_ARCTIC: return TREE_RANDOM_ARCTIC;
 		case LT_TROPIC:
 			switch (GetTropicZone(tile)) {
-				case TROPICZONE_NORMAL:  return (TreeType)(seed * TREE_COUNT_SUB_TROPICAL / 256 + TREE_SUB_TROPICAL);
-				case TROPICZONE_DESERT:  return (TreeType)((seed > 12) ? TREE_INVALID : TREE_CACTUS);
-				default:                 return (TreeType)(seed * TREE_COUNT_RAINFOREST / 256 + TREE_RAINFOREST);
+				case TROPICZONE_NORMAL:  return TREE_RANDOM_TROPIC_NORMAL;
+				case TROPICZONE_DESERT:  return ((seed > 12) ? TREE_INVALID : TREE_CACTUS);
+				default:                 return TREE_RANDOM_TROPIC_RAINFOREST;
 			}
+		default: return TREE_RANDOM_TOYLAND;
```

Comment:
While you're here, I think this would make more sense as an explicit case with a default NOT_REACHED() to follow most other OpenTTD code.